apps/protocol-frontend/src/components/DaoDashboardShell.tsx

```diff
@@ -10,17 +10,42 @@ import ContributionMembersPieShell from './ContributionMembersPieShell';
 import RecentContributionsTableShell from './RecentContributionsTableShell';
 import { subWeeks } from 'date-fns';
 import { YEAR, DEFAULT_DATE_RANGES } from '../utils/constants';
+import { useDaoUserUpdate } from '../hooks/useDaoUserUpdate';
+import { useUser } from '../contexts/UserContext';
+import { useContributionInfiniteList } from '../hooks/useContributionList';
+import { SortOrder } from '@govrn/protocol-client';
+import { mergePages } from '../utils/arrays';
+import { UIContribution } from '@govrn/ui-types';
+import { useNavigate } from 'react-router-dom';
+import GovrnAlertDialog from './GovrnAlertDialog';
+
 interface DaoDashboardShellProps {
   daoName: string;
   daoId: number;
 }
+
 const TODAY_DATE = new Date();
 
 const DaoDashboardShell = ({ daoName, daoId }: DaoDashboardShellProps) => {
-  const [showCustomDatePicker, setShowCustomDatePicker] = useState(false);
+  const navigate = useNavigate();
+  const { userData } = useUser();
+  const { mutateAsync: updateDaoMemberStatus, isLoading: isLeavingLoading } =
+    useDaoUserUpdate();
+  const { data, hasNextPage, fetchNextPage } = useContributionInfiniteList({
+    where: {
+      guilds: { some: { guild: { is: { id: { equals: daoId } } } } },
+    },
+    orderBy: { date_of_engagement: SortOrder.Desc },
+  });
+
+  const recentContributions = useMemo<UIContribution[]>(() => {
```

Comment:
Why move this up in to a higher component?

Reply:
Consider it three layer: 
1. `DaoDashboardShell`
2. `RecentContributionsTableShell`
3. `RecentContributionsTable`

1 -> 2: passes the `InfiniteData<>` object,
2 -> 3: passes 2D array: `InfiniteData.pages`. 

This is making types harder to read than using single array. 

So, moving it up would to be easier for future refactoring and building upon since data's type is same for all children and require unified type and less knowledge.